### ISSUE TITLE
Gecko IA2 virtual buffers: space is not inserted between nodes in some cases

### ISSUE THREAD

**Issue by michaelDCurran** [nvaccess/test_nvdaTracImport#149]
**Reported by pvagner on 21 Apr 2008 07:02 UTC**
When going to E.G. http://trac.nvda-project.org and navigating to the navigations list NVDA places all the list items in single line.
It looks as follows:
logged in as pvagnerLogoutPreferencesHelp/GuideAbout Trac 
I think this is not all in the single line. When activating virtual buffer pass through by pressing NVDA+space and trying to copy the whole page as firefox does it by pressing sequence of Ctrl+a and Ctrl+c it looks as follows:
     * logged in as pvagner
    * Logout
    * Preferences
    * Help/Guide
    * About Trac

I can understand that switching screen layout off makes these appear on seperate lines but that's because they are links not because they are really placed on seperate lines. Further more If this can't be made then it should be considered to seperate the links with spaces at least because now as NVDA presents it it's blended all together.

**Comment by michaelDCurran** [comment]
**Comment by jteh on 21 Apr 2008 07:21 UTC**
Always displaying list items on separate lines is not correct. List items can be displayed on the same line if the style is set accordingly; i.e. by setting display:inline on the li tags. See http://www.jantrid.net/ for an example of where this is done intentionally. Note that copying and pasting from Firefox does not seem to honour this style; even on this site, the list ends up vertical.

Trac is a bit strange. The stylesheet seems to indicate that the list should be displayed horizontally. However, checking this with another screen reader that provides video intercept shows that the navigation bar is not horizontal, although it has some other strange styling that makes it difficult to read with said video intercept.

In any case, the items definitely shouldn't be rendered without spaces between them. I'm not sure why this is occurring; spaces are correctly inserted in the case of http://www.jantrid.net/. Mick, do you have any idea why this might be occurring for certain sites such as Trac and Google?

**Comment by michaelDCurran** [comment]
**Comment by jteh on 21 Apr 2008 07:47 UTC**
Replying to [jteh](comment:1):
> Trac is a bit strange. The stylesheet seems to indicate that the list should be displayed horizontally. However, checking this with another screen reader that provides video intercept shows that the navigation bar is not horizontal...
Sighted assistance confirms that this is displayed horizontally after all.

**Comment by michaelDCurran** [comment]
**Modified by aleksey_s on 21 Apr 2008 07:57 UTC**

**Comment by michaelDCurran** [comment]
**Comment by pvagner on 21 Apr 2008 08:06 UTC**
We have came to the conclusion that list may be arranged horizontally and vertically. The trac issue has been verified by the sighted user. NVDA's behaviour is respecting the arrangement. To show the difference one can see some vertical lists at the page http://www.nvda-project.org/download.html
Finally it looks as the only problem is that sometimes no space or other delimiter is inserted between nodes. This trac nav bar is one such example and it can also be seen at the google.com main page where buttons "Google Search" and "I'm Feeling Lucky" are not delimited.

**Comment by michaelDCurran** [comment]
**Comment by jteh on 14 May 2008 05:40 UTC**
We rely on Gecko to include spacing in the text if appropriate. It is quite difficult for us to insert such spacing, as our text offsets need to follow Gecko's as closely as possible so that events work. This is arguably not a good idea anyway, as we should be following what Gecko gives us as closely as possible.

The question remains as to whether you are happy with the current situation. Although there is no space between nodes sometimes, moving by word now correctly stops on node boundaries. If you are happy with this, please close the ticket.

**Comment by michaelDCurran** [comment]
**Comment by pvagner on 14 May 2008 06:26 UTC**
Given all the obstacles it requires to workaround this issue and very likely to introduce new problems I think I am happy with the ability to stop on node boundaries while navigating by word.
Perhaps there is no space even from sighted person's view.

Ticket closed.
